### PR TITLE
Add new aws_cost_collector module

### PR DESCRIPTION
**Pull request by haggaret**
Add new aws_cost_collector module for getting data from the AWS Cost Explorer, instead of from cloudcheckr

I've sort of written each file (s3.py, etc) as a standalone unit that can be called, but they are also imported into aws_cost_collector and used from there. Can easily remove the main method if you don't like that/think it has any benefit.

### CODE REVIEW

File: team-cost-reporter/aws_cost_collector/aws_cost_collector.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    end_string = current_date_str
+    return (start_string,end_string)
+
+
+def load_data(service, days_to_report, cloudcheckr=False, debug=False):
+    start_date, end_date = get_start_end_strings(days_to_report)
+    group_by_tag = None
+    data_full = None
+    if service == 's3':
+        group_by_tag = 'signiant:email'
```
Comment (dnorth98) [inline team-cost-reporter/aws_cost_collector/aws_cost_collector.py]: Remove this.  The tags are in the config file.  There should be nothing Signiant specific in the code

File: team-cost-reporter/aws_cost_collector/aws_cost_collector.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        data_full = s3.get_costs(aws_cli_client=CE,
+                                 start_date=start_date,
+                                 end_date=end_date,
+                                 group_by_tag=group_by_tag,
+                                 next_token=None)
+    elif service == 'ecs':
+        group_by_tag = "aws:cloudformation:stack-name"
```
Comment (dnorth98) [inline team-cost-reporter/aws_cost_collector/aws_cost_collector.py]: Same as previous comment.  All tags should be in the config file

File: team-cost-reporter/aws_cost_collector/aws_cost_collector.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+        logging.error("Unknown service")
+
+    if cloudcheckr:
+        return (cloudcheckr_format.convert(data_full, group_by_tag))
+    return data_full
+
+
+if __name__ == "__main__":
+
+    LOG_FILENAME = 'aws_cost_collector.log'
```
Comment (dnorth98) [inline team-cost-reporter/aws_cost_collector/aws_cost_collector.py]: Should only log to STDOUT as this is designed to be run in a docker container

File: team-cost-reporter/aws_cost_collector/cloudcheckr_format.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                # Already have a dict entry for this tag_value - append another item to the list
+                tag_dict[tag_value].append(cost_entry)
+            else:
+                # No entry for this tag_value - start a new list
+                tag_dict[tag_value] = [cost_entry]
+
+    # At this point, tag_dict is a dict with keys for each unique tag_value
+    # eg.
+    # {
+    #     "signant:email user1@domain" : [
```
Comment (dnorth98) [inline team-cost-reporter/aws_cost_collector/cloudcheckr_format.py]: Signiant specific tag

File: team-cost-reporter/aws_cost_collector/cloudcheckr_format.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    #            "Amount": float,
+    #            "UsageQuantity": 0.0
+    #         },
+    #         {
+    #            "Date": "2018-01-02T00:00:00",
+    #            "Amount": float,
+    #            "UsageQuantity": 0.0
+    #         }
+    #     ],
+    #     "signant:email user2@domain" : [
```
Comment (dnorth98) [inline team-cost-reporter/aws_cost_collector/cloudcheckr_format.py]: Signiant specific tag

File: team-cost-reporter/aws_cost_collector/cloudcheckr_format.py [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    #         }
+    #     ]
+    # }
+    # Need the following format:
+    # Format:
+    # {
+    #    "Total": float, "Max": float, "Min": float, "Average": float,
+    #    "Groupings": [
+    #       {
+    #          "Name": "signiant:email user@domain.com",
```
Comment (dnorth98) [inline team-cost-reporter/aws_cost_collector/cloudcheckr_format.py]: Signiant specific tag

File: team-cost-reporter/aws_cost_collector/s3.py [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    return result
+
+
+if __name__ == "__main__":
+
+    LOG_FILENAME = 's3_cost_collector.log'
+
+    parser = argparse.ArgumentParser(description='aws_cost_collector')
+
+    parser.add_argument("--aws-access-key-id", help="AWS Access Key ID", dest='aws_access_key', required=False)
```
Comment (dnorth98) [inline team-cost-reporter/aws_cost_collector/s3.py]: Keys should be read from the environment, not passed in on the cli

File: team-cost-reporter/aws_cost_collector/s3.py [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+    parser.add_argument("--aws-secret-access-key", help="AWS Secret Access Key", dest='aws_secret_key', required=False)
+    parser.add_argument("--start-date", help="Start Date (YYYY-MM-DD)", dest='start_date', required=True)
+    parser.add_argument("--end-date", help="End Date (YYYY-MM-DD)", dest='end_date', required=True)
+    parser.add_argument("--cloudcheckr", help="Get data in Cloudcheckr format", action='store_true', required=False)
+    parser.add_argument("--region", help="The AWS region the cluster is in", dest='region', required=True)
+    parser.add_argument("--profile", help="The name of an aws cli profile to use.", dest='profile', default=None, required=False)
```
Comment (dnorth98) [inline team-cost-reporter/aws_cost_collector/s3.py]: Cli profile?  Are we just calling the CLI?

File: team-cost-reporter/aws_cost_collector/s3.py [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+    ch.setFormatter(console_formatter)
+    file_formatter = logging.Formatter('%(asctime)s - %(levelname)8s: %(message)s')
+    fh.setFormatter(file_formatter)
+    # Add the handlers to the logger
+    logger.addHandler(fh)
+    logger.addHandler(ch)
+
+    logging.info("Start")
+
+    SESSION = boto3.session.Session(profile_name=args.profile, region_name=args.region)
```
Comment (dnorth98) [inline team-cost-reporter/aws_cost_collector/s3.py]: Profile should be set in the environment

File: team-cost-reporter/plugins/s3/__init__.py [hunk 10]
```diff
@@ -20,40 +18,53 @@ def getTeamCost(team_name,configMap,debug):
...
+    data_url = None
+    aws_ce = False
     for config_plugin in configMap['plugins']:
         if config_plugin['name'] == id():
             if debug: log("plugin info found in config file")
-            data_url = config_plugin['data_url']
+            if 'data_url' in config_plugin:
+                data_url = config_plugin['data_url']
+            elif 'aws_ce' in config_plugin:
+                aws_ce = config_plugin['aws_ce']
```
Comment (dnorth98) [inline team-cost-reporter/plugins/s3/__init__.py]: else?  fail I guess....

**Review by dnorth98 — CHANGES_REQUESTED** [review]

**Comment by haggaret** [comment]
Ok, big changes made after @dnorth98's review. Have a look again Dave if you have a minute.

File: team-cost-reporter/aws_cost_collector.py [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+    # start=2016-06-01
+    # end=2016-06-30
+    start_string = n_days_ago_str
+    end_string = current_date_str
+    return (start_string,end_string)
+
+
+def get_costs(days_to_report, granularity, filter, group_by, next_token=None, debug=False):
+    SESSION = boto3.session.Session()
```
Comment (haggaret) [inline team-cost-reporter/aws_cost_collector.py]: Actually, I'm not crazy about this - I think I'll move this outside this function - doesn't make sense to be inside it.

**Review by dnorth98 — APPROVED** [review]
yeah, I like this much more.  Much simpler.

**Pull request merged**